core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -196,6 +197,18 @@ private SampleByIntervalIterator findSampleByIntervals(
             txnRangeLoader.load(engine, Path.PATH.get(), baseTableToken, txnIntervals, lastRefreshTxn, lastTxn);
             minTs = txnRangeLoader.getMinTimestamp();
             maxTs = txnRangeLoader.getMaxTimestamp();
+            // Check if refresh limit should be applied.
+            try (TableMetadata matViewMeta = engine.getTableMetadata(matViewToken)) {
+                final int limit = matViewMeta.getMatViewRefreshLimitHoursOrMonths();
+                if (limit != 0) {
+                    final long now = microsecondClock.getTicks();
```

Comment:
If the view is a sample by with a timezone, I believe we should measure 1 hour in that timezone, so that on shift back we either refresh last hour fully or don't refresh it at all

Reply:
The doc will say timestamps older than 1 hour, without mentioning time zones. As for gaps or backward clock shifts, `TimeZoneIntervalIterator` will take care of them by moving the low interval boundary to an earlier timestamp. So, I wouldn't make thing more complicated here.